File: src/main/java/ru/vood/enumFromTable/config/DataSourceConfiguration.java
package ru.vood.enumFromTable.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.jdbc.datasource.DataSourceTransactionManager;
import org.springframework.jdbc.datasource.DriverManagerDataSource;

@Configuration
public class DataSourceConfiguration {

    @Bean("driverManagerDataSource")
    public DriverManagerDataSource driverManagerDataSource() {

/*        Properties prop = new Properties();
        prop.put("driverClassName", "oracle.jdbc.OracleDriver");
        prop.put("username", "vood");
        prop.put("password","vood");*/

        DriverManagerDataSource driverManagerDataSource = new DriverManagerDataSource("jdbc:oracle:thin:@localhost:1521:vood"/*, prop*/);
        return driverManagerDataSource;
    }

    @Bean("dataSourceTransactionManager")
    public DataSourceTransactionManager dataSourceTransactionManager() {
        return new DataSourceTransactionManager(driverManagerDataSource());
    }

    @Bean("jdbcTemplateMy")
    public JdbcTemplate jdbcTemplate() {
        return new JdbcTemplate(driverManagerDataSource());
    }


}

File: src/main/java/ru/vood/enumFromTable/config/DataSourceConfiguration.kt
package ru.vood.enumFromTable.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.jdbc.datasource.DataSourceTransactionManager
import org.springframework.jdbc.datasource.DriverManagerDataSource
import java.util.*

@Configuration
open class DataSourceConfiguration {

    @Bean
    open fun driverManagerDataSource(): DriverManagerDataSource {
        val prop = Properties()
        prop["driverClassName"] = "oracle.jdbc.OracleDriver"
        prop["username"] = "vood"
        prop["password"] = "vood"
        return DriverManagerDataSource("jdbc:oracle:thin:@localhost:1521:vood", prop)
    }

/*
    @Bean
    open fun entityManagerFactory():EntityManagerFactory{

    }
*/

    @Bean
    open fun dataSourceTransactionManager(): DataSourceTransactionManager {
        return DataSourceTransactionManager(driverManagerDataSource())
    }

}